common.mk

```diff
@@ -355,7 +355,7 @@ buildroot: optee-os optee-rust
 		$(DEFCONFIG_FTPM) \
 		--br-defconfig out-br/extra.conf \
 		--make-cmd $(MAKE))
-	@$(MAKE) $(br-make-flags) -C ../out-br all
+	@export PATH=$${HOME}/.cargo/bin:$${PATH} && $(MAKE) $(br-make-flags) -C ../out-br all
```

Comment:
Why is this somewhat more complicated way of writing `@export PATH=$(HOME)/.cargo/bin:$(PATH) ...`?

By the way, can we rely on `$(HOME)/.cargo/bin` to contain the right thing? What if the user for some reason messes around with the user-specific cargo installation? This toolchain isn't special in any way compared to the GCC toolchain we're using and we're not expecting the user to keep a sane GCC cross-toolchain ready for us.

Reply:
> By the way, can we rely on `$(HOME)/.cargo/bin` to contain the right thing?

Yes, because that path is populated by the `optee-rust` target which is a dependency of this one.

Reply:
Do you mean that `optee-rust` should use a more specific path to install the toolchain?

Reply:
To keep stuff in `~/.cargo/bin/`, that is.

Reply:
> Do you mean that `optee-rust` should use a more specific path to install the toolchain?

I'm sorry, I missed your comment. Yes, I believe a more safe choice would be to install the rust toolchain under the `$(ROOT)/toolchains` directory.

Reply:
OK. Then I think this would require some changes in https://github.com/apache/incubator-teaclave-trustzone-sdk/blob/no-std/setup.sh in order to support choosing the installation directory. @b49020

Reply:
I have been thinking about moving Rust toolchain installation from `setup.sh` to rather `toolchain.mk` which aligns to the comments above. So I will work on a patch to remove `setup.sh` invocation and rather add Rust toolchain installation instructions to `toolchain.mk` instead.

Reply:
Also, as per https://stackoverflow.com/questions/57765424/overriding-rust-installation-default-paths-home-cargo-and-home-rustup, we can customize Rust toolchain installation for OP-TEE buildroot setup.